### DISCUSSION TITLE
Lost connection to the database

### DISCUSSION THREAD

**Discussion by revoluta** [godror/godror#350] [General]
Hello,
My problem is that I connect to an Oracle database in AWS that has been configured to cut off inactive connections every 350 seconds.

I don't know how to configure godror so that connections are not lost. I have tried the following, but it doesn't work. If I go a while without making requests and then I make a query, it no longer responds.

` func (s *ServerStruct) InitializeDataBase() {
	var err error

	StrCon := s.DbUser + "/" + s.DbPassWord + "@" + s.DbHost + ":" +
		strconv.Itoa(s.DbPort) + "/" + s.DbName

	s.Db, err = sql.Open("godror", StrCon)
	if err != nil {
		s.Log.Fatal("Cannot connect to database:", err)
	}

	s.Db.SetMaxOpenConns(100)
	s.Db.SetMaxIdleConns(10)
	s.Db.SetConnMaxLifetime(30 * time.Minute)
	s.Db.SetConnMaxIdleTime(300 * time.Second)

	if err = s.Db.Ping(); err != nil {
		s.Log.Fatal("Ping Error connecting to the database:", err)
	}

} `

**Comment by tgulacsi** [comment]
Try EXPIRE_TIME (https://download.oracle.com/ocomdocs/global/Oracle-Net-Easy-Connect-Plus.pdf)

**Reply by revoluta** [reply]
Thank you very much, friend, for the help, it solved my problem.
